components/panel/body.js

```diff
@@ -26,16 +26,22 @@ class PanelBody extends Component {
 
 	toggle( event ) {
 		event.preventDefault();
-		this.setState( ( state ) => ( {
-			opened: ! state.opened,
-		} ) );
+		if ( this.props.opened === undefined ) {
+			this.setState( ( state ) => ( {
+				opened: ! state.opened,
+			} ) );
+		}
+
+		if ( this.props.onToggle ) {
+			this.props.onToggle();
+		}
 	}
 
 	render() {
-		const { title, children } = this.props;
-		const { opened } = this.state;
-		const icon = `arrow-${ opened ? 'down' : 'right' }`;
-		const className = classnames( 'components-panel__body', { 'is-opened': opened } );
+		const { title, children, opened } = this.props;
+		const isOpened = opened === undefined ? this.state.opened : opened;
```

Comment:
Are there any places this component is used with the state variant now? Wonder if we should just treat it as a controlled component, rather than try to juggle both state and prop values of `opened`.

Reply:
Not right now, no strong preference.